components/configuration-accessor/default/package.mjs

```diff
@@ -16,51 +20,104 @@ const isPrefixArray = (prefix, array) => {
+    } else {
+      return index;
+    }
+  } else {
+    return null;
+  }
+};
+
+const executables = [
+  ["node"],
+  ["npm", "exec"],
+  ["npm", "x"],
+  ["npx"],
+  ["npm.cmd", "exec"],
+  ["npm.cmd", "x"],
+  ["npx.cmd"],
+  [],
+];
+
+export const splitTokens = (tokens) => {
+  const executable = executables.find((executable) =>
+    isPrefixArray(executable, tokens),
+  );
+  const positional = findPositionalIndex(tokens, executable.length);
+  assert(
+    !logErrorWhen(
+      positional === null,
+      "could not parse and hook command because of missing positional argument, got %j",
+      tokens,
     ),
-  ];
+    "could not parse and hook command",
+    ExternalAppmapError,
+  );
   return {
-    canParseSource: (source) => regexps.some((regexp) => regexp.test(source)),
-    parseSource: (source) => {
-      for (const regexp of regexps) {
-        const result = regexp.exec(source);
-        if (result !== null) {
-          return result.groups;
-        }
-      }
-      throw new InternalAppmapError("could not parse command source");
-    },
+    __proto__: null,
+    before: tokens.slice(0, positional + 1),
+    after: tokens.slice(positional + 1),
   };
 };
 
-export const generateSplitTokens = (name) => {
-  const prefixes = [
-    [name],
-    ["npx", name],
-    ["npx.cmd", name],
-    ["npm", "exec", name],
-    ["npm.cmd", "exec", name],
-  ];
+export const sniffTokens = (tokens, name) => {
+  const executable = executables.find((executable) =>
+    isPrefixArray(executable, tokens),
+  );
+  const positional = findPositionalIndex(tokens, executable.length);
+  return positional !== null && tokens[positional].includes(name);
+};
+
+const chopShell = (tokens) => {
+  let index = 0;
+  while (
+    index < tokens.length &&
+    !tokens[index].startsWith("-") &&
+    tokens[index].includes("=")
+  ) {
+    index += 1;
+  }
   return {
-    canSplitTokens: (tokens) =>
-      prefixes.some((prefix) => isPrefixArray(prefix, tokens)),
-    splitTokens: (tokens) => {
-      for (const prefix of prefixes) {
-        if (isPrefixArray(prefix, tokens)) {
-          return {
-            __proto__: null,
-            before: prefix,
-            after: tokens.slice(prefix.length),
-          };
-        }
-      }
-      throw new InternalAppmapError("could not split command tokens");
-    },
+    head: tokens.slice(0, index),
+    body: tokens.slice(index),
+  };
+};
+
+const chopCmdShell = (tokens) => ({
+  head: [],
+  body: tokens,
+});
+
+const isCmdShell = (shell) => shell === "cmd" || shell === "cmd.exe";
+
+export const parseSource = (source, shell) => {
+  const tokens = isCmdShell(shell)
+    ? tokenizeCmdShell(source)
+    : tokenizeShell(source);
+  const { head, body } = isCmdShell(shell)
+    ? chopCmdShell(tokens)
+    : chopShell(tokens);
+  const { before, after } = splitTokens(body);
+  return {
+    __proto__: null,
+    before: [...head, ...before].join(" "),
+    after: after.join(" "),
```

Comment:
I know naming things is difficult, but I think this would be much clearer if these fields were called `command` and `args` (what) instead of `before` and `after` (how). Confusing things further, before and after suggest there is something in the middle, and also the words could have a temporal meaning.

I know perhaps I'm nitpicking a bit, but I'm just trying to imagine what would have made reading, understanding and reviewing this code easier. I like to think these changes would also make maintaining it in the future easier.

Reply:
Nah it is all good. It seems obvious to me now but I know I will look at this in 6 months and WTF myself. 09f78516def2e50c6b02bf2b31ac2edec590745e